### DISCUSSION TITLE
Virtual-Voice-Assisant

### DISCUSSION THREAD

**Discussion by IanDublew** [openai/whisper#1106] [Show and tell]
Project Link: https://github.com/IanDublew/Virtual-Voice-Assisant
This project uses three different technologies to transcribe speech to text, search text query using GPT and give out output and then convert the output to speech:

Whisper: Whisper is a general-purpose speech recognition model. It is trained on a large dataset of diverse audio and is also a multitasking model that can perform multilingual speech recognition, speech translation, and language identification.

CHATGPT: GPT-3 is a state-of-the-art language model developed by OpenAI.

gTTS (Google Text-to-Speech) library gTTS is a Python library and CLI tool that uses Google Text-to-Speech API to convert text to speech. This library supports several languages and offers various customization options such as voice, speed, and pitch. For this project, we used gTTS to convert the transcribed text to speech and saved the audio as an MP3 file.

**Comment by jakenuts** [comment]
So ready for this to be. Open source no snooping super-Alexa is gonna happen, might as well be today.

**Reply by IanDublew** [reply]
It sure will.

**Reply by ventz** [reply]
Only if you can replace the "gTTS". Otherwise you are just swapping out "little snooper" for "much larger snooper" ;)

Jokes aside - awesome project!

There are plenty of libraries you can utilize for text to speech in an offline manner by the way, without having to hit an API. Some of them are pretty good too.  ex: `pyttsx3`

**Reply by IanDublew** [reply]
Thank you for your comments and suggestions! You make an excellent point about replacing gTTS with another library. I'll definitely check into the pyttsx3 library you recommended for offline text-to-speech capability.
